source/components/ch4_component.cpp

```diff
@@ -104,25 +104,22 @@ void CH4Component::setData( const string& varName,
     try {
          if( varName ==  D_PREINDUSTRIAL_CH4  ) {
             H_ASSERT( data.date == Core::undefinedIndex() , "date not allowed" );
-            M0 = unitval::parse_unitval( data.value_str, data.units_str, U_PPBV_CH4 );
```

Comment:
Hey @bpbond, all my PRs come from @rgieseke 's and my work on a Python and a JS wrapper for hector (https://github.com/swillner/pyhector). I do not what to change your system, only make changes that help a wrapper or other API users ;). This PR now allows sending a `unitval` initialized with a double to a component's `setData` (most of these currently assume the values to be in string). Concerning checks for units situation is the same:
In rc1.2, a value could be given without unit in a string and is equipped with the proper unit in `unitval::parse_unitval` (or, with a uit given, the unit is checked).
In the PR, a value can be given as either string or double (with unit U_UNDEFINED or given) and is equipped with the proper unit in `unitval::set`/`tseries::set` (or unit checked there).
Advantage of the string/double-agnostic `setData` is improved performance, e.g. when providing emission data as number array by preventing unnecessary string<->double conversions. Also, then, there is a single point in allowing/disallowing undefined units (`unitval::expecting_unit`).
All in all, thanks for your wonderful and fast model!